lib/logstash/outputs/elasticsearch_java.rb

```diff
@@ -328,122 +305,70 @@ def register
     @retry_queue_not_full = ConditionVariable.new
     @retry_queue = Queue.new
 
-    client_settings = {}
-
-
-    if @protocol.nil?
-      @protocol = LogStash::Environment.jruby? ? "node" : "http"
-    end
-
-    if @protocol == "http"
-      if @action == "create_unless_exists"
-        raise(LogStash::ConfigurationError, "action => 'create_unless_exists' is not supported under the HTTP protocol");
-      end
 
-      client_settings[:path] = "/#{@path}/".gsub(/\/+/, "/") # Normalize slashes
-      @logger.debug? && @logger.debug("Normalizing http path", :path => @path, :normalized => client_settings[:path])
+    if @protocol ==='node' && !@network_host
+      raise LogStash::ConfigurationError, "network_host MUST be set if the 'node' protocol is in use!"
     end
 
-    if ["node", "transport"].include?(@protocol)
-      # Node or TransportClient; requires JRuby
-      raise(LogStash::PluginLoadingError, "This configuration requires JRuby. If you are not using JRuby, you must set 'protocol' to 'http'. For example: output { elasticsearch { protocol => \"http\" } }") unless LogStash::Environment.jruby?
-
-      client_settings["cluster.name"] = @cluster if @cluster
-      client_settings["network.host"] = @bind_host if @bind_host
-      client_settings["transport.tcp.port"] = @bind_port if @bind_port
-      client_settings["client.transport.sniff"] = @sniffing
- 
-      if @node_name
-        client_settings["node.name"] = @node_name
-      else
-        client_settings["node.name"] = "logstash-#{Socket.gethostname}-#{$$}-#{object_id}"
-      end
+    client_settings = {}
+    client_settings["cluster.name"] = @cluster if @cluster
```

Comment:
gotcha, we can make it an explicit `:default => "elasticsearch"` later (not required for this PR)